pkgs/servers/sql/mysql/8.0.x.nix

```diff
@@ -45,6 +45,7 @@ self = stdenv.mkDerivation rec {
     "-DWITH_ARCHIVE_STORAGE_ENGINE=yes"
     "-DWITH_BLACKHOLE_STORAGE_ENGINE=yes"
     "-DWITH_FEDERATED_STORAGE_ENGINE=yes"
+    "-DWITH_ROUTER=no" ## mysqlrouter is new addition in 8.0 source-tree, but it currently fails to build on Darwin (tested: macOS 10.12.6 + mysql 8.0.17)
```

Comment:
Given this is only failing on `darwin` you could make conditional on `stdenv.isDarwin`, if you were so inclined.

Reply:
If it's dealer's-choice, then I'd lean toward clean "off" since I don't use `mysqlrouter` but do rely on portability between nix's Linux/Darwin environments. But that opinion isn't super-strong - if it's general-practice round here to do a toggle like that, then I'm happy to put the toggle in.

Reply:
dealer's choice it is :+1: